### PR TITLE
Bael 2328

### PR DESCRIPTION
**Pull request by MherBaghinyan**

### CODE REVIEW

File: java-strings/src/main/java/com/baeldung/string/MatchWords.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class MatchWords {
+
+    public static boolean containsWordsIndexOf(String inputString, String[] words) {
+        boolean found = true;
+        for (String word : words) {
+            int index = inputString.indexOf(word);
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/MatchWords.java]: No need to assign a variable here, just use if(inputString.indexOf(word) == -1)

File: java-strings/src/main/java/com/baeldung/string/MatchWords.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        Pattern pattern = Pattern.compile(regexp.toString());
+        if (pattern.matcher(inputString).find()) {
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean containsWordsJava8(String inputString, String[] words) {
+        List inputStringList = Arrays.asList(inputString.split(" "));
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/MatchWords.java]:
```suggestion
        List<String> inputStringList = Arrays.asList(inputString.split(" "));
```

File: java-strings/src/main/java/com/baeldung/string/MatchWords.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        List<String> wordsList = Arrays.asList(words);
+
+        return wordsList.stream().allMatch(inputStringList::contains);
+    }
+
+    public static boolean containsWordsArray(String inputString, String[] words) {
+        List inputStringList = Arrays.asList(inputString.split(" "));
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/MatchWords.java]:
```suggestion
        List<String> inputStringList = Arrays.asList(inputString.split(" "));
```

File: java-strings/src/main/java/com/baeldung/string/MatchWords.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    public static boolean containsWordsPatternMatch(String inputString, String[] words) {
+
+        StringBuilder regexp = new StringBuilder();
+        for (String word : words) {
+            regexp.append("(?=.*").append(word).append(")");
+        }
+
+        Pattern pattern = Pattern.compile(regexp.toString());
+        if (pattern.matcher(inputString).find()) {
```
Comment (KevinGilmore) [inline java-strings/src/main/java/com/baeldung/string/MatchWords.java]: Simplify: return pattern.matcher(inputString).find());

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+public class MatchWordsUnitTest {
+
+    private final String[] words = {"hello", "Baeldung"};
+    private final String inputString = "hello there, Baeldung";
+    private final String wholeInput = "helloBaeldung";
+
+    @Test
+    public void givenText_whenCallingStringContains_shouldMatchWords() {
+        final boolean result = MatchWords.containsWords(inputString, words);
+        assertThat(result).isEqualTo(true);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isTrue();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingJava8_shouldMatchWords() {
+        final boolean result = MatchWords.containsWordsJava8(inputString, words);
+        assertThat(result).isEqualTo(true);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isTrue();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingJava8_shouldNotMatchWords() {
+        final boolean result = MatchWords.containsWordsJava8(wholeInput, words);
+        assertThat(result).isEqualTo(false);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isFalse();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingPattern_shouldMatchWords() {
+        final boolean result = MatchWords.containsWordsPatternMatch(inputString, words);
+        assertThat(result).isEqualTo(true);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isTrue();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingAhoCorasick_shouldMatchWords() {
+        final boolean result = MatchWords.containsWordsAhoCorasick(inputString, words);
+        assertThat(result).isEqualTo(true);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isTrue();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingAhoCorasick_shouldNotMatchWords() {
+        final boolean result = MatchWords.containsWordsAhoCorasick(wholeInput, words);
+        assertThat(result).isEqualTo(false);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isFalse();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingIndexOf_shouldMatchWords() {
+        final boolean result = MatchWords.containsWordsIndexOf(inputString, words);
+        assertThat(result).isEqualTo(true);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isTrue();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingArrayList_shouldMatchWords() {
+        final boolean result = MatchWords.containsWordsArray(inputString, words);
+        assertThat(result).isEqualTo(true);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isTrue();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void givenText_whenCallingArrayList_shouldNotMatchWords() {
+        final boolean result = MatchWords.containsWordsArray(wholeInput, words);
+        assertThat(result).isEqualTo(false);
```
Comment (KevinGilmore) [inline java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]:
```suggestion
        assertThat(result).isFalse();
```

File: java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java (hunk 5 above)
Comment (eric-martin) [reply java-strings/src/test/java/com/baeldung/string/MatchWordsUnitTest.java]: I will make these changes in another PR (no perms to push changes)

**Pull request merged**